presto-main/src/main/java/com/facebook/presto/sql/planner/PlannerUtils.java

```diff
@@ -420,4 +440,28 @@ public static RowExpression coalesce(List<RowExpression> expressions)
     {
         return new SpecialFormExpression(SpecialFormExpression.Form.COALESCE, expressions.get(0).getType(), expressions);
     }
+
+    public static boolean isSupportedArrayContainsFilter(FunctionResolution functionResolution, RowExpression filterExpression, List<VariableReferenceExpression> left, List<VariableReferenceExpression> right)
+    {
+        List<Type> supportedTypes = ImmutableList.of(BIGINT, INTEGER, VARCHAR, DATE);
+
+        if (filterExpression instanceof CallExpression) {
+            CallExpression callExpression = (CallExpression) filterExpression;
+            if (functionResolution.isArrayContainsFunction(callExpression.getFunctionHandle())) {
+                RowExpression array = callExpression.getArguments().get(0);
```

Comment:
I think the PushDownFilterExpressionEvaluationThroughCrossJoin should already handle the pushdown of array expression, and array should be VariableReferenceExpression here?

Reply:
fixed in the check below